libs/plugin-cache/src/lib/provider.ts

```diff
@@ -0,0 +1,7 @@
+import { HttpExtResponse } from '@http-ext/http-ext';
+
+export interface CacheProvider {
```

Comment:
Not sure if the URL is enough. We will at least need the query params.
Maybe some other use cases might need other data. I don't know... we've got to think about this.
Maybe the key is a hash generated using some `request => hash` customizable function.